experiments/DaggerInstanceGLA/src/DaggerInstanceGLA.jl

```diff
@@ -0,0 +1,78 @@
+module DaggerInstanceGLA
+
+import Base: +
+using AutoHashEquals
+
+export DagDom, LinearMap, MatrixThunk, matrixToThunk,
+       dom, codom, adjoint, +, compose, id, oplus, mzero,
+       braid, mcopy, delete, plus, zero, scalar, antipode,
+       ⊗
+using Dagger
+using LinearAlgebra
+using Catlab.LinearAlgebra.GraphicalLinearAlgebra
+using Catlab, Catlab.Doctrines
+import ...Doctrines:
+  Ob, Hom, dom, codom, compose, ⋅, ∘, id, oplus, ⊕, mzero, braid,
+  dagger, dunit, dcounit, mcopy, Δ, delete, ◊, mmerge, ∇, create, □,
+  plus, zero, coplus, cozero, meet, top, join, bottom
+using LinearMaps
+import LinearMaps: adjoint
+const LMs = LinearMaps
+
+
+@auto_hash_equals struct DagDom
+  N::Int
+end
+
+# This structure was created to keep track of dom and codom information.
+# This information can be updated efficiently, and keeping it here keeps
+# LinearFunctions from having to think the thunk each time the dom or codom
+# is queried
+
+struct MatrixThunk
+  thunk::Thunk
+  dom::Int
+  codom::Int
+end
+
+matrixToThunk(A::LinearMap) = begin
+  delayed(x->x)(A)
+end
+
+@instance LinearFunctions(DagDom, MatrixThunk) begin
+
+  adjoint(f::MatrixThunk) = MatrixThunk(delayed(adjoint)(f.thunk), f.codom, f.dom)
+  +(f::MatrixThunk, g::MatrixThunk) = MatrixThunk(delayed(+)(f.thunk, g.thunk), f.dom, f.codom)
+
+  dom(f::MatrixThunk) = f.dom
+  codom(f::MatrixThunk) = f.codom
+
+  compose(f::MatrixThunk, g::MatrixThunk) = MatrixThunk(delayed(*)(g.thunk,f.thunk), g.dom, f.codom)
+  id(V::DagDom) = MatrixThunk(delayed(x->x)(LMs.UniformScalingMap(1, V.N)), V.N, V.N)
+
+  oplus(V::DagDom, W::DagDom) = DagDom(V.N + W.N)
+  oplus(f::MatrixThunk, g::MatrixThunk) = MatrixThunk(delayed(LMs.BlockDiagonalMap)(f.thunk, g.thunk), f.dom+g.dom, f.codom+g.codom)
+  mzero(::Type{DagDom}) = DagDom(0)
+  braid(V::DagDom, W::DagDom) =
+  MatrixThunk(delayed(x->x)(LinearMap(braid_lm(V.N), braid_lm(W.N), W.N+V.N, V.N+W.N)),V.N+W.N, W.N+V.N)
+
+  mcopy(V::DagDom)  = MatrixThunk(delayed(x->x)(LinearMap(mcopy_lm, plus_lm, 2*V.N, V.N)), V.N, 2*V.N)
```

Comment:
I think these should be `MatrixThunk(delayed(x->LinearMap(mcopy_lm, plus_lm, 2*V.N, V.N)*x), V.N, 2*V.N)` etc.

Reply:
Yeah, I see what you're getting at with that, I had a hard time with this one (as well as braid and delete). I think as-is will cause a typeError, since `delayed(x->LinearMap(mcopy_lm, plus_lm, 2*V.N, V.N)*x)` is not of type `Thunk`.

From the way it's used in the LinearMaps instance, `mcopy` returns a morphism (LinearMap object) which is then applied to other morphisms through composition. If we're keeping with that usage (applying `mcopy` by composition), I think we'll want this to return a MatrixThunk which has an identity Thunk for the LinearMap `mcopy`, which can then be used as an argument in a `compose` call.